### PR TITLE
Fix diagnostics in Objective-C documentation comments appearing on the wrong line

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://129668876

## Summary

Now that Clang and Swift emits source ranges for documentation comments the same, we need to remove the DocC workaround that offset Objective-C ranges, otherwise diagnostics for Objective-C documentation comments will appear on the wrong line. 

## Dependencies

None

## Testing

Add some content that will result in warnings to an Objective-C documentation comment and build documentation for it. For example:

```objc
#import <Foundation/Foundation.h>

NS_ASSUME_NONNULL_BEGIN

/// Some description of my class
///
/// Can link to ``doSomething`` but not to ``NotFoundSymbol`` or <doc:NotFoundArticle>.
///
/// @Metadata {
///    @DisplayName("MyClass")
/// }
@interface MyClass : NSObject

/// Some description of some function
///
/// - Parameter something: This method has no parameters.
/// - Returns: This method doesn't return anything
- (void)doSomething;

@end

NS_ASSUME_NONNULL_END
```

The various warnings should display for the correct line and column ranges.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] ~Updated documentation if necessary~ Not applicable

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Tests/SwiftDocCTests/Test Bundles/ObjCFrameworkWithInvalidLink.docc/symbol-graphs/clang/HelloWorldFramework.symbols.json [hunk 1]
```diff
@@ -67,12 +67,12 @@
...
               "end": {
-                "character": 52,
-                "line": 10
+                "character": 51,
+                "line": 9
               },
               "start": {
-                "character": 5,
-                "line": 10
+                "character": 4,
```
Comment (anferbui) [inline Tests/SwiftDocCTests/Test Bundles/ObjCFrameworkWithInvalidLink.docc/symbol-graphs/clang/HelloWorldFramework.symbols.json:74]: Did you edit this symbol graph manually or did you generate it from some Objective-C source code?

**Review by anferbui — APPROVED** [review]
Small question about the test but LGTM!

File: Tests/SwiftDocCTests/Test Bundles/ObjCFrameworkWithInvalidLink.docc/symbol-graphs/clang/HelloWorldFramework.symbols.json (hunk 1 above)
Comment (d-ronnqvist) [reply Tests/SwiftDocCTests/Test Bundles/ObjCFrameworkWithInvalidLink.docc/symbol-graphs/clang/HelloWorldFramework.symbols.json:74]: Manually. I don't know if a source for this test content exists (anymore).

**Pull request merged**